icechunk/src/ops/gc.rs

```diff
@@ -321,3 +321,124 @@ async fn gc_snapshots(
         .boxed();
     Ok(storage.delete_snapshots(storage_settings, to_delete).await?)
 }
+
+/// Expire snapshots older than a threshold.
+///
+/// This only processes snapshots found by navigating `reference`
+/// ancestry. Any other snapshots are not touched.
+///
+/// The operation will edit in place the oldest non-expired snapshot,
+/// changing its parent to be the root of the repo.
```

Comment:
This means the initial commit snapshot is always the parent right?

Reply:
Exactly, yes, the root of the repo will become the new parent of the first non-expired snapshot